### PR TITLE
Better CI pipeline

### PR DESCRIPTION
**Pull request by wyzlle**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,25 +94,32 @@ jobs:
...
+      
       - name: Install cargo-nextest
         uses: taiki-e/install-action@nextest
 
       - name: Run cargo nextest
         uses: actions-rs/cargo@v1
         with:
           command: nextest
-          args: run
+          args: run --package nixpacks --test generate_plan_tests
```
Comment (coffee-cup) [inline .github/workflows/ci.yml:105]: Should this also run the module unit tests?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,25 +94,32 @@ jobs:
...
 
-  lints:
-    name: Lints
+  test-docker:
+    needs: test-plan
+    name: Docker Tests
     runs-on: ubuntu-latest
-    steps:
+    strategy:
+      matrix: ${{fromJson(needs.test-plan.outputs.matrix)}}
```
Comment (coffee-cup) [inline .github/workflows/ci.yml:116]: 👌

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -53,25 +94,32 @@ jobs:
...
       - name: Run cargo nextest
         uses: actions-rs/cargo@v1
         with:
           command: nextest
-          args: run
+          args: run --package nixpacks --test generate_plan_tests
+      - name: List docker tests (prep test-docker)
+        id: docker-prep
+        run: |
+          echo "::set-output name=matrix::{\"tests\":[\"$(cargo test -q --package nixpacks --test docker_run_tests -- --list --format=terse | sed -z 's/: test\n/\", \"/g' | sed 's/...$//')]}"
```
Comment (coffee-cup) [inline .github/workflows/ci.yml:109]: For future selves, maybe a good idea adding a comment explaining wtf this is doing

**Review by coffee-cup — APPROVED** [review]
Nice 👍 
Very excited to have these tests run faster. One small question about running the unit tests but besides that it looks good.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wyzlle) [reply .github/workflows/ci.yml]: Whoops, missed `--lib` argument

File: .github/workflows/ci.yml (hunk 3 above)
Comment (wyzlle) [reply .github/workflows/ci.yml:110]: Done!

**Pull request merged**